windows/hid.c

```diff
@@ -590,11 +599,18 @@ static void hid_internal_get_ble_info(struct hid_device_info* dev, DEVINST dev_n
 	}
 }
 
-static void hid_internal_get_info(const wchar_t* interface_path, struct hid_device_info* dev)
+/* unfortunately, BLUETOOTH vs BLE can't be distinguished using hid_device_info alone
+   so we'll return it as a flag in the high 32-bit (low 32-bits will carry the DEVINST)
+*/
+
+#define HID_INTERNAL_INFO_FLAG_BLE 0x100000000ULL
+
+static ULONGLONG hid_internal_get_info(const wchar_t* interface_path, struct hid_device_info* dev)
```

Comment:
this is no longer `hid_internal_get_info`
lets name it explicitly: `hid_internal_get_bus_type`

Reply:
Makes sense. Also, I refactored the internal BLE flag processing to avoid OR-ing it with a `DEVINST`. Also, if you don't mind, let's call the function `hid_internal_detect_bus_type`, not `hid_internal_get_bus_type`  (because it returns `DEVINST`, not `hid_bus_type`). 

Take a look.

Reply:
if we're to bring this conversation, to my taste - both are hacks
I'd rather make a separate struct with DEVINST+flags as separate members (as an out parameter)

as for the suggestion naming - I'm fine with it

Reply:
Fair enough; let's return flags explicitly. `hid_internal_detect_bus_type` now returns a struct with two fields (`dev_node` and `bus_flags`).